### DISCUSSION TITLE
Exif orientation causes error when used with sequential access

### DISCUSSION THREAD

**Discussion by tonymarklove** [libvips/libvips#3402] [General]
## Bug report

**Describe the bug**
JPEG images with an Exif orientation of LeftBottom cause an error (sometimes?) when used with sequential access.

**To Reproduce**
I'm using ruby-vips, but I'm assuming the same problem would be seen in the C library.

1. I run the following code:
```ruby
image = Vips::Image.new_from_file('exif-orientated-left-bottom.jpg', access: :sequential)
image = image.autorot
image.jpegsave_buffer
```
2. Use the image attached below (grabbed from [here](https://github.com/recurser/exif-orientation-examples/blob/master/Landscape_8.jpg)):
![exif-orientated-left-bottom](https://user-images.githubusercontent.com/45162/226945864-b37f3c51-d80a-418a-878e-25f06fb7bf22.jpg)

I was thinking that this might be expected behaviour, but when combined with certain additional transformations, libvips seems to correct itself. But I apologize if this isn't actually a bug. For example the following works without error:
```ruby
image = Vips::Image.new_from_file('exif-orientated-left-bottom.jpg', access: :sequential)
image = image.autorot
image = image.thumbnail_image(200, size: :down)
image.jpegsave_buffer
```

**Expected behavior**
The image should be saved without error.

**Actual behavior**
There are a bunch of warnings printed:
```
(process:13303): VIPS-WARNING **: 14:47:41.342: error in tile 0 x 256

(process:13303): VIPS-WARNING **: 14:47:41.342: error in tile 0 x 264

...
```
And eventually an out-of-order read error:
```
VipsJpeg: out of order read at line 1800
```

**Environment**
 - OS: MacOS 13.2.1
 - Vips: 8.14.1

**Comment by jcupitt** [comment]
Hi @tonymarklove,

Yes, this is expected behaviour.

I'd recommend:

```
image = Vips::Image.thumbnail('exif-orientated-left-bottom.jpg', 200, size: :down)
image.jpegsave_buffer
```

It'll do the rotate for you automatically, and it's 10x faster than `thumbnail_image` (which is only there for emergencies and should not be used).

**Reply by tonymarklove** [reply]
Interesting. Thanks!

My full use-case performs a bunch of different transforms, which can all be selected by end users, so I don't know ahead of time which ones will be used:

```ruby
image = Vips::Image.new_from_file(file.path, access: :sequential)
image = image.autorot
image = flip_horizontally(image)
image = crop(image)
image = resize_to_limit(image)
image = composite_overlay(image)
image = transform_colorspace(image, file_type)
```

Where each of those method calls will do something or nothing based on configuration. So I don't think `Vips::Image.thumbnail` will help me here unfortunately. For now I'm just planning to switch back from `sequential` to `random` access mode, which will at least prevent errors even if it's not the most efficient approach.
